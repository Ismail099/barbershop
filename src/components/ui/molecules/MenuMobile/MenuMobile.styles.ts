import styled, { css } from 'styled-components';

import { MenuMobileProps } from './MenuMobile';

type OpenMenuMobileProps = Pick<MenuMobileProps, 'isOpen'>;

export const Wrapper = styled.div<OpenMenuMobileProps>`
  ${({ theme, isOpen }) => css`
    background: ${theme.colors.darkGray};
    opacity: ${isOpen ? 0.9 : 0};
    pointer-events: ${isOpen ? 'all' : 'none'};
    transition: 0.3s linear;
    position: fixed;
    top: 0;
    left: 0;
    bottom: 0;
    right: calc(4.8rem + (1.5rem + 1.5rem));
    z-index: ${theme.layers.menu};
    color: white;
    display: flex;
    align-items: center;
    justify-content: center;
    overflow-y: auto;
    overflow-x: hidden;
  `}
`;

export const WrapperList = styled.ul`
  ${({ theme }) => css`
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    gap: ${theme.spacings.medium};
  `}
`;

export const List = styled.li`
  ${({ theme }) => css`
    position: relative;
    list-style: none;
    text-align: center;
    display: flex;
    justify-content: center;

    &:first-child::after {
      width: 100%;
    }

    &:first-child {
      color: ${theme.colors.primary};
      text-shadow: 0px 0px 30px rgba(247, 24, 18, 0.5), 0px 0px 15px rgba(247, 24, 18, 0.4),
        0px 0px 10px rgba(247, 24, 18, 0.4), 0px 0px 5px rgba(247, 24, 18, 0.2);
    }

    &::after {
      content: '';
      position: absolute;
      justify-self: center;
      bottom: -0.5rem;
      width: 0;
      height: 0.2rem;
      background: ${theme.colors.primary};
      box-shadow: 0px 0px 10px ${theme.colors.primary};
      transition: 0.3s linear;
    }

    &:hover::after,
    &:focus::after {
      width: 100%;
    }
  `}
`;

export const Link = styled.a`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.xmedium};
    font-weight: 300;
    padding: 0 1rem;
    transition: 0.3s linear;

    &:hover,
    &:focus {
      color: ${theme.colors.primary};
      cursor: pointer;
      text-shadow: 0px 0px 30px rgba(247, 24, 18, 0.5), 0px 0px 15px rgba(247, 24, 18, 0.4),
        0px 0px 10px rgba(247, 24, 18, 0.4), 0px 0px 5px rgba(247, 24, 18, 0.2);
    }
  `}
`;
