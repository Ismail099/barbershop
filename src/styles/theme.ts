export default {
  grid: {
    container: '129rem'
  },
  border: {
    radius: '0.4rem'
  },
  font: {
    family: {
      primary: 'Dancing Script',
      secundary: 'Abril Fatface',
      tertiary:
        "Poppins, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif"
    },
    thin: 300,
    light: 400,
    regular: 500,
    medium: 600,
    bold: 700,
    sizes: {
      xsmall: '1.2rem',
      small: '1.4rem',
      medium: '1.6rem',
      xmedium: '1.8rem',
      xxmedium: '2.0rem',
      big: '2.2rem',
      xbig: '2.5rem',
      xxbig: '3.0rem',
      large: '3.5rem',
      xlarge: '4.5rem',
      xxlarge: '5.0rem',
      extra: '6.0rem'
    }
  },
  colors: {
    primary: '#F71812',
    secondary: '#1923F6',
    mainBg: '#333333',
    white: '#FFFFFF',
    lightRed: '#F17575',
    lightGray: '#EEEEEE',
    gray: '#181818',
    darkGray: '#080808',
    black: '#000000'
  },
  spacings: {
    xxsmall: '0.8rem',
    xsmall: '1.6rem',
    small: '2.4rem',
    medium: '3.2rem',
    large: '4.0rem',
    xlarge: '4.8rem',
    xxlarge: '5.6rem'
  },
  layers: {
    base: 10,
    menu: 20,
    overlay: 30,
    modal: 40,
    alwaysOnTop: 50
  }
};
