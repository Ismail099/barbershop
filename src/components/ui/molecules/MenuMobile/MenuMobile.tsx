import * as S from './MenuMobile.styles';

export type MenuMobileProps = {
  isOpen: boolean;
};

export const MenuMobile = ({ isOpen }: MenuMobileProps) => {
  return (
    <S.Wrapper aria-hidden={!isOpen} aria-label="Menu Full" isOpen={isOpen}>
      <S.WrapperList>
        <S.List>
          <S.Link aria-label="Home">Home</S.Link>
        </S.List>
        <S.List>
          <S.Link aria-label="Sobre">Sobre</S.Link>
        </S.List>
        <S.List>
          <S.Link aria-label="Serviços">Serviços</S.Link>
        </S.List>
        <S.List>
          <S.Link aria-label="Equipe">Equipe</S.Link>
        </S.List>
        <S.List>
          <S.Link aria-label="Opinião">Opinião</S.Link>
        </S.List>
        <S.List>
          <S.Link aria-label="FeedBack">FeedBack</S.Link>
        </S.List>
      </S.WrapperList>
    </S.Wrapper>
  );
};
