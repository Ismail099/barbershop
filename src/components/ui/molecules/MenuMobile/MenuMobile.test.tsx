import { screen } from '@testing-library/react';
import theme from '~src/styles/theme';
import { renderWithTheme } from '~src/utils/tests/helpers';

import { MenuMobile } from './MenuMobile';

describe('<MenuMobile />', () => {
  it('Deve renderizar os links do componente <MenuMobile />', () => {
    const { container } = renderWithTheme(<MenuMobile isOpen={true} />);

    expect(screen.getByLabelText(/Home/i)).toBeInTheDocument();
    expect(screen.getByLabelText(/Sobre/i)).toBeInTheDocument();
    expect(screen.getByLabelText(/Serviços/i)).toBeInTheDocument();
    expect(screen.getByLabelText(/Equipe/i)).toBeInTheDocument();
    expect(screen.getByLabelText(/Opinião/i)).toBeInTheDocument();
    expect(screen.getByLabelText(/FeedBack/i)).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('Deve mudar a cor do link para a cor principal quando tiver Hover no componente <MenuMobile />', () => {
    renderWithTheme(<MenuMobile isOpen={true} />);

    expect(screen.getByLabelText(/Sobre/i)).toHaveStyleRule('color', theme.colors.primary, {
      modifier: ':hover'
    });
  });
});
