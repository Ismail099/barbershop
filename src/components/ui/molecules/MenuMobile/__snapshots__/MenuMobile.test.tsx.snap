// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuMobile /> Deve renderizar os links do componente <MenuMobile /> 1`] = `
.c0 {
  background: #080808;
  opacity: 0.9;
  pointer-events: all;
  -webkit-transition: 0.3s linear;
  transition: 0.3s linear;
  position: fixed;
  top: 0;
  left: 0;
  bottom: 0;
  right: calc(4.8rem + (1.5rem + 1.5rem));
  z-index: 20;
  color: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  overflow-y: auto;
  overflow-x: hidden;
}

.c1 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 3.2rem;
}

.c2 {
  position: relative;
  list-style: none;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2:first-child::after {
  width: 100%;
}

.c2:first-child {
  color: #F71812;
  text-shadow: 0px 0px 30px rgba(247,24,18,0.5),0px 0px 15px rgba(247,24,18,0.4),0px 0px 10px rgba(247,24,18,0.4),0px 0px 5px rgba(247,24,18,0.2);
}

.c2::after {
  content: '';
  position: absolute;
  justify-self: center;
  bottom: -0.5rem;
  width: 0;
  height: 0.2rem;
  background: #F71812;
  box-shadow: 0px 0px 10px #F71812;
  -webkit-transition: 0.3s linear;
  transition: 0.3s linear;
}

.c2:hover::after,
.c2:focus::after {
  width: 100%;
}

.c3 {
  font-size: 1.8rem;
  font-weight: 300;
  padding: 0 1rem;
  -webkit-transition: 0.3s linear;
  transition: 0.3s linear;
}

.c3:hover,
.c3:focus {
  color: #F71812;
  cursor: pointer;
  text-shadow: 0px 0px 30px rgba(247,24,18,0.5),0px 0px 15px rgba(247,24,18,0.4),0px 0px 10px rgba(247,24,18,0.4),0px 0px 5px rgba(247,24,18,0.2);
}

<div
  aria-hidden="false"
  aria-label="Menu Full"
  class="c0"
>
  <ul
    class="c1"
  >
    <li
      class="c2"
    >
      <a
        aria-label="Home"
        class="c3"
      >
        Home
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        aria-label="Sobre"
        class="c3"
      >
        Sobre
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        aria-label="Serviços"
        class="c3"
      >
        Serviços
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        aria-label="Equipe"
        class="c3"
      >
        Equipe
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        aria-label="Opinião"
        class="c3"
      >
        Opinião
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        aria-label="FeedBack"
        class="c3"
      >
        FeedBack
      </a>
    </li>
  </ul>
</div>
`;
